irc/getters.go

```diff
@@ -5,13 +5,15 @@ package irc
 
 import (
 	"github.com/oragono/oragono/irc/isupport"
+	"github.com/oragono/oragono/irc/languages"
 	"github.com/oragono/oragono/irc/modes"
 )
 
-func (server *Server) Config() *Config {
+func (server *Server) Config() (config *Config) {
 	server.configurableStateMutex.RLock()
-	defer server.configurableStateMutex.RUnlock()
-	return server.config
+	config = server.config
+	server.configurableStateMutex.RUnlock()
+	return
```

Comment:
Just... thinking about this since it's popped up here, is this operation actually safe? We're returning a pointer to `server.config`, is there an opportunity for `server` to modify `server.config` after it's been returned, or is it not a worry since `server.config` itself is static and it's only ever replaced wholesale?

Reply:
Correct, the config object is never modified after initial load.

In theory this could just be [atomic.LoadPointer](https://golang.org/pkg/sync/atomic/#LoadPointer) but unfortunately that means giving up type safety.